src/TokenHandler/AccountManager.mo

```diff
@@ -110,14 +128,44 @@ module {
 
       if (latestDeposit < prevDeposit) freezeCallback("latestDeposit < prevDeposit on notify");
 
+      let depositInfo = depositRegistry.get(p);
+
+      // precredit deposit funds
       if (latestDeposit > fee_) {
-        // precredit deposit funds
-        ignore creditRegistry.credit(p, latestDeposit - fee_);
+        if (depositInfo.dust == 0) {
```

Comment:
We can check this on the fly as `prevDeposit > fee_`, don't need the `depositInfo.dust` field. See PR where I removed it. Better to keep the balance information in a single place rather than two places.